### ISSUE TITLE
Inclusion in scikit-mine

### ISSUE THREAD

**Issue by remiadon** [chuanconggao/PrefixSpan-py#33]
Hi there, very nice and rich implementation of these 3 algorithms

The INRIA center at Rennes is creating a new python library, namely [scikit-mine](https://github.com/scikit-mine/scikit-mine), to centralise pattern mining methods, and improve inter-operability and consistency with other fields, such as Machine Learning.

Your API already has similarities with what scikit-mine provides, being:
- [an implementation of LCM](https://scikit-mine.github.io/scikit-mine/reference/preprocessing.html#lcm) for closed frequent itemsets (equivalent to BIDE, but for regular itemsets)
- [custom callbacks](https://scikit-mine.github.io/scikit-mine/tutorials/callbacks/callbacks.html) for richer exploration

In the context of scikit-mine, only BIDE and FEAT would be nice to have, as PrefixSpan mines too many patterns, and we encourage concise representations.

I also plan to try FEAT as a candidate generator for [SQS-candidates](https://eda.mmci.uni-saarland.de/pubs/2012/the_long_and_the_short_of_it-tatti,vreeken.pdf), an algorithm based on MDL.
To this purpose handling gaps would be required, as SQS natively accounts them when running its optimization process

Anyone to provide support for integration into scikit-mine ?

**Comment by chuanconggao** [comment]
Hi, it is an interesting idea.

As the maintainer, I may provide some help but it requires some effort estimation. If there any planning or scope so far?

**Comment by remiadon** [comment]
@chuanconggao thanks for responding,

I think we can start with something "simple" 
 1. integrating BIDE, w.r.t functional definition in skmine
 2. unit tests for BIDE
 3. adding relevant example / doc for it

Rough estimation for this would be 2 weeks (my side) with daily feedback from you

Once this is done, I would add another task 
- make sure the skmine.callback API is consistent with your callback API (we can treat similar use cases with both)

**Comment by chuanconggao** [comment]
Sounds good.

On my side, I can start some refactoring to make it more robust.

**Comment by remiadon** [comment]
@chuanconggao I wrapped your code and added a few unit tests in the PR mentioned above

What sort of refactoring were you thinking of ?

My implem is still missing the top-k part though ... which is interesting to have

**Comment by chuanconggao** [comment]
The refactoring part is mostly docstring, typing, and bug fixes.
